### PR TITLE
Bug 1855696 - Update FxA strings

### PR DESCRIPTION
**Pull request by t-p-white**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1855696

### CODE REVIEW

File: fenix/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -478,11 +478,15 @@
...
+    <string name="preferences_override_fxa_server" moz:RemovedIn="120" tools:ignore="UnusedResources">Custom Firefox Account server</string>
+    <!-- Preference to override the Mozilla Account server -->
+    <string name="preferences_override_account_server">Custom Mozilla Account server</string>
     <!-- Preference to override the Sync token server -->
     <string name="preferences_override_sync_tokenserver">Custom Sync server</string>
     <!-- Toast shown after updating the FxA/Sync server override preferences -->
-    <string name="toast_override_fxa_sync_server_done">Firefox Account/Sync server modified. Quitting the application to apply changes…</string>
+    <string name="toast_override_fxa_sync_server_done" moz:RemovedIn="120" tools:ignore="UnusedResources">Firefox Account/Sync server modified. Quitting the application to apply changes…</string>
+    <!-- Toast shown after updating the Mozilla Account/Sync server override preferences -->
+    <string name="toast_override_account_sync_server_done">Mozilla Account/Sync server modified. Quitting the application to apply changes…</string>
```
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:489]: @Delphine I don't think we usually want our Mozilla tradework exposed to be translated. Is this also the case here?

File: fenix/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -478,11 +478,15 @@
     <!-- Preference for accessibility -->
     <string name="preferences_accessibility">Accessibility</string>
     <!-- Preference to override the Firefox Account server -->
-    <string name="preferences_override_fxa_server">Custom Firefox Account server</string>
+    <string name="preferences_override_fxa_server" moz:RemovedIn="120" tools:ignore="UnusedResources">Custom Firefox Account server</string>
+    <!-- Preference to override the Mozilla Account server -->
+    <string name="preferences_override_account_server">Custom Mozilla Account server</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:483]: "account" should almost always be lowercase, except in "“Mozilla Account Privacy Policy” and “Mozilla Account Terms of Service” (anytime other navigation items are using headline casing, for consistency)
Content guidelines: https://www.figma.com/file/7nsOBSt3VTeCimDqavQT9J/Mozilla-Accounts-Content-Guidelines?type=design&node-id=1903-403&mode=design&t=gtC3OzSNcm5JK5b5-0

File: fenix/app/src/main/res/values/strings.xml (hunk 1 above)
Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:489]: We didn't parameterize "Firefox accounts" as per branding guidelines in the past, so we can keep doing the same for "Mozilla account" (again, no capitalization on "account")

File: fenix/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -498,7 +502,9 @@
     <!-- Preference description for banner about signing in -->
     <string name="preferences_sign_in_description_2">Sign in to sync tabs, bookmarks, passwords, and more.</string>
     <!-- Preference shown instead of account display name while account profile information isn't available yet. -->
-    <string name="preferences_account_default_name">Firefox Account</string>
+    <string name="preferences_account_default_name" moz:RemovedIn="120" tools:ignore="UnusedResources">Firefox Account</string>
+    <!-- Preference shown instead of account display name while account profile information isn't available yet. -->
+    <string name="preferences_account_default_name_2">Mozilla Account</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:507]: "account" should be lowercase

**Review by Delphine — CHANGES_REQUESTED** [review]
Made a few comments, after double checking against content guidelines found [here](https://www.figma.com/file/7nsOBSt3VTeCimDqavQT9J/Mozilla-Accounts-Content-Guidelines?type=design&node-id=1903-403&mode=design&t=gtC3OzSNcm5JK5b5-0).

I wasn't sure at first if these strings should be changed to simply "accounts" (vs "Mozilla accounts"), but since these strings reference the account server (vs a user's account), I think using the branding here makes sense.

**Comment by t-p-white** [comment]
Thanks @Delphine, have updated with the suggested changes

**Review by gabrielluong — APPROVED** [review]

**Review by Delphine — APPROVED** [review]
strings.xml section LGTM

**Pull request merged**